relayer/relays/execution/message/message.go

```diff
@@ -0,0 +1,320 @@
+				}).Info("last synced execution block unchanged")
+			}
+
+			select {
+			case <-ctx.Done():
+				return nil
+			case <-ticker.C:
+				continue
+			}
+		}
+	})
+
+	return nil
+}
+
+func (m *Message) syncBasic(ctx context.Context, eg *errgroup.Group, secondLastSyncedBlockNumber, lastSyncedBlockNumber uint64) error {
+	addressNonceMap := make(map[common.Address]uint64, len(m.addresses))
+	for _, address := range m.addresses {
+		addressNonceMap[address] = 1
+	}
+
+	log.WithFields(log.Fields{
+		"start": secondLastSyncedBlockNumber,
+		"end":   lastSyncedBlockNumber,
+	}).Info("fetching basic channel messages")
+	basicPayload, err := m.listener.ProcessBasicEvents(ctx, secondLastSyncedBlockNumber, lastSyncedBlockNumber, addressNonceMap)
+	if err != nil {
+		return err
+	}
+
+	return m.writeBasicMessages(ctx, basicPayload)
+}
+
+func (m *Message) syncIncentivized(ctx context.Context, eg *errgroup.Group, secondLastSyncedBlockNumber, lastSyncedBlockNumber uint64) error {
+	log.WithFields(log.Fields{
+		"start": secondLastSyncedBlockNumber,
+		"end":   lastSyncedBlockNumber,
+	}).Info("fetching incentivized channel messages")
+
+	incentivizedPayload, err := m.listener.ProcessIncentivizedEvents(ctx, secondLastSyncedBlockNumber, lastSyncedBlockNumber)
+	if err != nil {
+		return err
+	}
+
+	return m.writeIncentivizedMessages(ctx, incentivizedPayload)
+}
+
+func (m *Message) writeBasicMessages(ctx context.Context, payload ParachainPayload) error {
+	log.WithField("count", len(payload.Messages)).Info("writing basic messages")
+
+	for _, msg := range payload.Messages {
+		err := m.writer.WriteToParachainAndWatch(ctx, msg.Call, msg.Args...)
+		if err != nil {
+			return err
+		}
+
+		err = m.checkMessageVerificationResult(msg.Origin, msg.Nonce)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (m *Message) checkMessageVerificationResult(msgAddress common.Address, msgNonce uint64) error {
+	nonce, err := m.writer.GetLastBasicChannelNonceByAddress(msgAddress)
+	if err != nil {
+		return fmt.Errorf("fetch last basic channel message nonces by addresses: %w", err)
+	}
+
+	if nonce != msgNonce {
+		return fmt.Errorf("last basic message verification failed for address %s (expected nonce: %d, actual nonce: %d)", msgAddress, msgNonce, nonce)
+	}
+
+	log.WithFields(log.Fields{"nonce": msgNonce, "address": msgAddress}).Info("basic message verified successfully")
+	return nil
+}
+
+func (m *Message) writeIncentivizedMessages(ctx context.Context, payload ParachainPayload) error {
+	log.WithField("count", len(payload.Messages)).Info("writing incentivized messages")
+
+	for _, msg := range payload.Messages {
+		err := m.writer.WriteToParachainAndWatch(ctx, msg.Call, msg.Args...)
+		if err != nil {
+			return err
+		}
+
+		lastNonce, err := m.writer.GetLastIncentivizedChannelNonce()
+		if err != nil {
+			return err
+		}
+
+		if lastNonce != msg.Nonce {
+			return fmt.Errorf("last incentivized message verification failed (nonce: %d)", lastNonce)
```

Comment:
Let's include the expected nonce, like with the basic channel verification:

```suggestion
			return fmt.Errorf("last incentivized message verification failed (expected nonce: %d, actual nonce: %d)", msg.Nonce, lastNonce)
```